### PR TITLE
try not to raise Exceptions to avoid premature exit of long testrun t…

### PR DESCRIPTION
**Pull request by pruan-rht**
…o allow tests to continue instead of exiting

Tried with two separate run 20220524-1826 & 20220524-2345 and get the same number of waiting cases w/o seeing premature exit.  We can either wait for more sample runs or merge as is.

### CODE REVIEW

File: lib/polarshift/test_record.rb [hunk 1]
```diff
@@ -173,9 +173,11 @@ def update_to!(new_state)
       def start_scenario_for!(test_case)
         scenario = self.test_case.scenarios.find { |s| s.match! test_case}
         unless scenario
-          raise "logic error: trying to start Cucumber scenario not part of this test record"
+          puts "logic error: trying to start Cucumber scenario not part of this test record\n"
```
Comment (jhou1) [inline lib/polarshift/test_record.rb:176]: Shall we use `logger.warn` instead of `puts`?

File: lib/polarshift/test_suite.rb [hunk 2]
```diff
@@ -108,7 +108,8 @@ def commit!(test_case)
         if self.current_test_record.start_scenario_for! test_case
           return true
         else
-          raise "logic error: starting test case that is not part of current test record"
+          puts "logic error: starting test case that is not part of current test record\n"
```
Comment (jhou1) [inline lib/polarshift/test_suite.rb:111]: We can add the test case id here, for info and reference in failure analysis

Comment (pruan-rht) [reply lib/polarshift/test_suite.rb]: done

File: lib/polarshift/test_record.rb (hunk 1 above)
Comment (pruan-rht) [reply lib/polarshift/test_record.rb]: done...

File: lib/polarshift/test_suite.rb [hunk 3]
```diff
@@ -108,7 +108,8 @@ def commit!(test_case)
         if self.current_test_record.start_scenario_for! test_case
           return true
         else
-          raise "logic error: starting test case that is not part of current test record"
+          puts "logic error: starting test case '#{test_case}' that is not part of current test record\n"
```
Comment (jhou1) [inline lib/polarshift/test_suite.rb:111]: @pruan-rht should the `puts` also be replaced with `logger`?

Comment (pruan-rht) [reply lib/polarshift/test_suite.rb:111]: @jhou1 there is no logger defined in the file

Comment (pruan-rht) [reply lib/polarshift/test_suite.rb:111]: we can add the logger class at a later time

Comment (jhou1) [reply lib/polarshift/test_suite.rb:111]: OK, thanks

**Pull request merged**